lib/logstash/inputs/beats/message_listener.rb

```diff
@@ -107,5 +111,18 @@ def from_filebeat?(hash)
     def from_logstash_forwarder?(hash)
       !hash[LSF_LOG_LINE_FIELD].nil?
     end
+
+    def increment_connection_count
+      current_connection_count = @connections_list.size
+      @metric.gauge(:current_connections, current_connection_count)
+      if current_connection_count > @peak_connection_count.value
+        @peak_connection_count.value = current_connection_count
+        @metric.gauge(:peak_connections, @peak_connection_count.value)
+      end
+    end
```

Comment:
We can add a simple unit test for this? Since the `MessageListener` is self contained?